app/src/main/java/com/jerboa/ui/components/common/MarkdownHelper.kt

```diff
@@ -0,0 +1,116 @@
+            .usePlugin(CoilImagesPlugin.create(context, loader))
+            .build()
+    }
+
+    @Composable
+    fun CreateMarkdownView(
+        markdown: String,
+        color: Color = Color.Unspecified,
+        onClick: (() -> Unit)? = null,
+        onLongClick: (() -> Unit)? = null,
+    ) {
+        val style = MaterialTheme.typography.bodyLarge
+        val defaultColor: Color = LocalContentColor.current.copy(alpha = LocalContentAlpha.current)
+
+        AndroidView(
+            factory = { ctx ->
+                createTextView(
+                    context = ctx,
+                    color = color,
+                    defaultColor = defaultColor,
+                    fontSize = TextUnit.Unspecified,
+                    style = style,
+                    viewId = null,
+                    onClick = onClick,
+                    onLongClick = onLongClick,
+                )
+            },
+            update = { textView ->
+                markwon!!.setMarkdown(textView, markdown)
+//            if (disableLinkMovementMethod) {
+//                textView.movementMethod = null
+//            }
+            },
+        )
+    }
+
+    private fun createTextView(
+        context: Context,
+        color: Color = Color.Unspecified,
+        defaultColor: Color,
+        fontSize: TextUnit = TextUnit.Unspecified,
+        textAlign: TextAlign? = null,
+        maxLines: Int = Int.MAX_VALUE,
+        @FontRes fontResource: Int? = null,
+        style: TextStyle,
+        @IdRes viewId: Int? = null,
+        onClick: (() -> Unit)? = null,
+        onLongClick: (() -> Unit)? = null,
+    ): TextView {
+        val textColor = color.takeOrElse { style.color.takeOrElse { defaultColor } }
+        val mergedStyle = style.merge(
+            TextStyle(
+                color = textColor,
+                fontSize = if (fontSize != TextUnit.Unspecified) fontSize else style.fontSize,
+                textAlign = textAlign,
+            ),
+        )
+        return TextView(context).apply {
+            onClick?.let { setOnClickListener { onClick() } }
+            onLongClick?.let { setOnLongClickListener { onLongClick(); true } }
+            setTextColor(textColor.toArgb())
+            setMaxLines(maxLines)
+            setTextSize(TypedValue.COMPLEX_UNIT_DIP, mergedStyle.fontSize.value)
+            width = maxWidth
+
+            viewId?.let { id = viewId }
+            textAlign?.let { align ->
+                textAlignment = when (align) {
+                    TextAlign.Left, TextAlign.Start -> View.TEXT_ALIGNMENT_TEXT_START
+                    TextAlign.Right, TextAlign.End -> View.TEXT_ALIGNMENT_TEXT_END
+                    TextAlign.Center -> View.TEXT_ALIGNMENT_CENTER
+                    else -> View.TEXT_ALIGNMENT_TEXT_START
+                }
+            }
+
+            fontResource?.let { font ->
+                typeface = ResourcesCompat.getFont(context, font)
+            }
+        }
+    }
+}
```

Comment:
Not the hugest fan of all this custom code, rather than just using a decent jetpack-compose markdown library. But if there are no real good ones, I suppose we don't have a choice.

Reply:
I am not aware of one. The one you worked on previously does have a PR from two weeks ago with some possible fixes however I am not sure how well maintained that project is.